src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -317,7 +319,28 @@ public void respawn()
 	@Override
 	public void setGameMode(@NotNull GameMode mode)
 	{
-		gamemode = mode;
+		PlayerGameModeChangeEvent gameModeChangeEvent = new PlayerGameModeChangeEvent(this, mode);
+
+		ServerMock server = MockBukkit.getMock();
+		server.getPluginManager().callEvent(gameModeChangeEvent);
+
+		if (!gameModeChangeEvent.isCancelled())
+		{
+			gamemode = mode;
+		}
+	}
+	public @NotNull PlayerGameModeChangeEvent simulateChangeGameMode(@NotNull GameMode mode)
+	{
+		PlayerGameModeChangeEvent gameModeChangeEvent = new PlayerGameModeChangeEvent(this, mode);
+
+		ServerMock server = MockBukkit.getMock();
+		server.getPluginManager().callEvent(gameModeChangeEvent);
```

Comment:
No need to get the Server instance statically here, we can just use `Entity#getServer()`

```suggestion
		getServer().getPluginManager().callEvent(gameModeChangeEvent);
```